plugins/inputs/exec/run_notwinodws.go

```diff
@@ -0,0 +1,54 @@
+//go:build !windows
+
+package exec
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	osExec "os/exec"
+	"runtime"
+	"syscall"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/kballard/go-shellquote"
+)
+
+func (c CommandRunner) Run(
+	command string,
+	environments []string,
+	timeout time.Duration,
+) ([]byte, []byte, error) {
+	splitCmd, err := shellquote.Split(command)
+	if err != nil || len(splitCmd) == 0 {
+		return nil, nil, fmt.Errorf("exec: unable to parse command: %w", err)
+	}
+
+	cmd := osExec.Command(splitCmd[0], splitCmd[1:]...)
+	if runtime.GOOS != "windows" {
+		cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	}
```

Comment:
Hmmm why did you split the code? Just to set the process attribute? If so, why not use the code of `run_notwindows.go` with the `runtime.GOOS` statement? I think that's the only difference between the two...

Reply:
Unfortunately, Windows' `SysProcAttr` does not have the `Setpgid` attribute that we set here, so builds will fail. An Linux's `SysProcAttr` does not have the `CreationFlags`

[linux SysProcAttr](https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/syscall/exec_linux.go;l=62)
[winodows SysProcAttr](https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/syscall/exec_windows.go)

Reply:
Ok, but then you should remove the `if runtime.GOOS != "windows" {` part for the non-Windows files... :-)

Reply:
ah yes :) done!